pilot/pkg/model/sidecar.go

```diff
@@ -173,7 +194,10 @@ func ConvertToSidecarScope(ps *PushContext, sidecarConfig *Config, configNamespa
 	}
 
 	r := sidecarConfig.Spec.(*networking.Sidecar)
-	out := &SidecarScope{}
+	out := &SidecarScope{
+		configDependencies:    make(map[resource.GroupVersionKind]map[string]struct{}),
+		namespaceDependencies: make(map[string]struct{}),
```

Comment:
I think the problem comes from namespace is not a type of config. As we can see from the code below, the value of namespace changes comes from "Service.Attributes.Namespace". It's also not a part of the config. There are namespace dependency declarations in SidecarScope which not associate with any kind of configs. 
```
s.EnvoyXdsServer.ConfigUpdate(&model.PushRequest{
	Full:              true,
	NamespacesUpdated: map[string]struct{}{si.Service.Attributes.Namespace: {}},
	// TODO: extend and set service instance type, so no need re-init push context
	ConfigsUpdated: map[resource.GroupVersionKind]map[string]struct{}{model.ServiceEntryKind: {}},
	reason:         []model.TriggerReason{model.ServiceUpdate},
})
```

Reply:
Yeah. `namespaceDependencies` is an impl we used to have. 

Do u remember that we've discussed this above? According to my understanding `namespace` is not a kind of config and if we merge it into `configDependencies` we may need to use some tricky ways such as `ConfigKey{Namespace: ns}` and leave `Kind` and `Name` field to be empty. But it's hard to say what an empty `Kind` or `Name` means.

Reply:
Why do we even need to track a "namespace changed" at all though? I think it's just a legacy.. even in the example above we are not saying the namespace changed, we are saying the service IN the namespace changed

Reply:
@howardjohn Did you remember this PR is based on #22495. We can take iterations like that. I totally agree that we should remove the legacy "namespace changed" if we can, but is used to be there. We need to do some code refactoring and this really involves further work instead of full push scoping.

Please don't worry, the work of next PR will follow this.

Reply:
Yes. I see that the `namespaceDependencies` is just derived from the depended services. Thus if we describe the `configUpdated` and `configDependency` more detailed, this field can be removed. Let the whole work be done together in the PR next week, ok?